cuDSS/simple_matrix_market/simple_matrix_market.cpp

```diff
@@ -0,0 +1,470 @@
+                         status, "cudssMatrixCreateDn for b");
+    CUDSS_CALL_AND_CHECK(cudssMatrixCreateDn(&x, nrows, nrhs, ldx, x_values_d, CUDA_R_64F,
+                                             CUDSS_LAYOUT_COL_MAJOR),
+                         status, "cudssMatrixCreateDn for x");
+
+    /* Create a matrix object for the sparse input matrix. */
+    cudssMatrix_t A;
+    
+    printf("--- Starting resolution and timing --- \n");
+    cudaEventRecord(start);
+    cudssIndexBase_t base = CUDSS_BASE_ZERO;
+    CUDSS_CALL_AND_CHECK(cudssMatrixCreateCsr(&A, nrows, ncols, nnz, csr_offsets_d, NULL,
+                                              csr_columns_d, csr_values_d, CUDA_R_32I, CUDA_R_64F, mtype, mview,
+                                              base),
+                         status, "cudssMatrixCreateCsr");
+    cudaEventRecord(stop);
+    cudaEventSynchronize(stop);
+    cudaEventElapsedTime(&time_ms, start, stop);
+    printf("cuDSS MatrixCreateCsr time: %.4f ms\n", time_ms);
+    total_ms += time_ms;
+
+    /* Symbolic factorization */
+    cudaEventRecord(start);
+    CUDSS_CALL_AND_CHECK(cudssExecute(handle, CUDSS_PHASE_ANALYSIS, solverConfig, solverData,
+                                      A, x, b),
+                         status, "cudssExecute for analysis");
+    cudaEventRecord(stop);
+    cudaEventSynchronize(stop);
+    cudaEventElapsedTime(&time_ms, start, stop);
+    printf("cuDSS Analysis time: %.4f ms\n", time_ms);
+    total_ms += time_ms;
+
+    /* Factorization */
+    cudaEventRecord(start);
+    CUDSS_CALL_AND_CHECK(cudssExecute(handle, CUDSS_PHASE_FACTORIZATION, solverConfig,
+                                      solverData, A, x, b),
+                         status, "cudssExecute for factor");
+    cudaEventRecord(stop);
+    cudaEventSynchronize(stop);
+    cudaEventElapsedTime(&time_ms, start, stop);
+    printf("cuDSS Factorization time: %.4f ms\n", time_ms);
+    total_ms += time_ms;
+
+    /* Solving */
+    cudaEventRecord(start);
+    CUDSS_CALL_AND_CHECK(cudssExecute(handle, CUDSS_PHASE_SOLVE, solverConfig, solverData,
+                                      A, x, b),
+                         status, "cudssExecute for solve");
+    cudaEventRecord(stop);
+    cudaEventSynchronize(stop);
+    cudaEventElapsedTime(&time_ms, start, stop);
+    printf("cuDSS Solve time: %.4f ms\n", time_ms);
+    total_ms += time_ms;
+
+
+    /* Print the solution and compare against the exact solution */
+    CUDA_CALL_AND_CHECK(cudaMemcpy(x_values_h, x_values_d, nrhs * n * sizeof(double), cudaMemcpyDeviceToHost), "cudaMemcpy for x_values");
+    
+    double residual;
+
+    residual = compute_residual_error(n, csr_offsets_h, csr_columns_h, csr_values_h, x_values_h, b_values_h, mview);
+    
+    printf("cuDSS Total time: %.4f ms\n", total_ms);
+    printf("--- Resolution over ! --- \n");
+
+    printf("Residual L2 error ||Ax - b|| = %e\n", residual);
+    bool passed = (residual < 1e-5);
```

Comment:
nit: I would swap the print order (saying that it is over first, then list the total time, followed by computing the residual norm).
If you changed the print above to `Start solving`, this one should then read `Solving complete`.